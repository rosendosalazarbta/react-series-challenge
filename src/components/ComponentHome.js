import React, { Component, Fragment, } from 'react';
import ComponentGif from './ComponentGif';
import { connect } from 'react-redux';
import { fetchedGifsTrending, fetchedGifsSearch, } from '../actions/gifActions';
import { addGifToFavorites, removeGifToFavorites, } from '../actions/gifsFavorites';

class ComponentHome extends Component {
    constructor(props) {
        super(props);
    }

    componentDidMount() {
        this.props.dispatch(fetchedGifsTrending());
    }

    render() {
        console.log('favorites', this.props.favorites);
        return (
            <div>
                <div style={{ marginBottom: 10, }}>
                    <input type="text" placeholder="search" onKeyUp={this.onKeyUp} />
                </div>
                {this.renderLoader()}
                <Fragment>
                    {this.renderGifs()}
                </Fragment>
            </div>
        )
    }

    renderGifs = () => {
        return this.props.gifs.map(x => {
            const isFavorite = this.props.favorites.find(y => y.id == x.id);
            return ComponentGif(x.id, x.title, x.gif, isFavorite, () => {
                this.props.dispatch(addGifToFavorites(x));
            });
        });
    }

    renderLoader = () => {
        return (!this.props.loading ? null :
            <h4 className="text-center">Loading...</h4>
        );
    }

    onKeyUp = (e) => {
        if (e.keyCode == 13) {
            this.props.dispatch(fetchedGifsSearch(e.target.value));
        }
    }
}

const mapStateToProps = (state) => {
    return {
        gifs: state.gifsTrending.gifs,
        favorites: state.gifsFavorites.favorites,
        loading: state.gifsTrending.loading,
    };
};

export default connect(mapStateToProps)(ComponentHome);