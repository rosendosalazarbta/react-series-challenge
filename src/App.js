import React, { Component } from 'react';
import { Link, Redirect, Route, Switch, } from 'react-router-dom';
import ComponentHome from './components/ComponentHome';
import ComponentFavorites from './components/ComponentFavorites';
import './styles/challenge.css';

class App extends Component {
  render() {
    return (
      <div className="app">
        <header className="navbar">
          <div className="navbar-container">
            <ul>
              <li>
                <Route path="/Home" children={({ match }) => {
                  const active = match ? 'active' : false;
                  return <Link to="/Home" className={active}>Home</Link>;
                }} />
              </li>
              <li>
                <Route path="/Favorites" children={({ match }) => {
                  const active = match ? 'active' : false;
                  return <Link to="/Favorites" className={active}>Favorites</Link>;
                }} />
              </li>
            </ul>
          </div>
        </header>
        <Switch>
          <Route path="/Home" component={ComponentHome} />
          <Route path="/Favorites" component={ComponentFavorites} />
          <Redirect to={{ pathname: "/Home", }} />
        </Switch>
      </div >
    );
  }
}

export default App;
