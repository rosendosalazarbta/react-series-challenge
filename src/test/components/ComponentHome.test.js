import React from 'react';
import ComponentHome from '../../components/ComponentHome';
import configureStore from 'redux-mock-store';
import { ContainerInputText, InputText, H4 } from '../../styles/App.style';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

const initialState = {
  gifsTrending: {
    gifs: [],
    loading: true,
  },
  gifsFavorites: {
    favorites: [],
    textFilter: '',
  },
};
const mockStore = configureStore();
const store = mockStore(initialState);
describe('ComponentHome test', () => {

  test('Should match snapshot', () => {
    const favComponent = renderer.create(<ComponentHome store={store} />).toJSON();
    expect(favComponent).toMatchSnapshot();
  });

  test('Should render ContainerInputText', () => {
    const favComponent = mount(<ComponentHome store={store} />);
    expect(favComponent.find(ContainerInputText).length).toEqual(1);
  });

  test('Should render InputText', () => {
    const favComponent = mount(<ComponentHome store={store} />);
    expect(favComponent.find(InputText).length).toEqual(1);
  });

  test('Should render LOADING', () => {
    const favComponent = mount(<ComponentHome store={store} />);
    expect(favComponent.find(H4).length).toEqual(1);
  });
});
