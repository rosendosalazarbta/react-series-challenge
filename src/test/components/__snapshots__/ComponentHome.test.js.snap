// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Home Should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(ComponentHome)
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "runSaga": Object {
          "@@redux-saga/TASK": true,
          "cancel": [Function],
          "cont": undefined,
          "done": Promise {},
          "error": [Function],
          "id": 1,
          "isAborted": [Function],
          "isCancelled": [Function],
          "isRunning": [Function],
          "joiners": Array [],
          "name": "root",
          "result": [Function],
          "setContext": [Function],
        },
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "dispatch": [Function],
      "favorites": Array [],
      "gifs": Array [],
      "loading": true,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "runSaga": Object {
          "@@redux-saga/TASK": true,
          "cancel": [Function],
          "cont": undefined,
          "done": Promise {},
          "error": [Function],
          "id": 1,
          "isAborted": [Function],
          "isCancelled": [Function],
          "isRunning": [Function],
          "joiners": Array [],
          "name": "root",
          "result": [Function],
          "setContext": [Function],
        },
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "runSaga": Object {
            "@@redux-saga/TASK": true,
            "cancel": [Function],
            "cont": undefined,
            "done": Promise {},
            "error": [Function],
            "id": 1,
            "isAborted": [Function],
            "isCancelled": [Function],
            "isRunning": [Function],
            "joiners": Array [],
            "name": "root",
            "result": [Function],
            "setContext": [Function],
          },
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "dispatch": [Function],
        "favorites": Array [],
        "gifs": Array [],
        "loading": true,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "runSaga": Object {
            "@@redux-saga/TASK": true,
            "cancel": [Function],
            "cont": undefined,
            "done": Promise {},
            "error": [Function],
            "id": 1,
            "isAborted": [Function],
            "isCancelled": [Function],
            "isRunning": [Function],
            "joiners": Array [],
            "name": "root",
            "result": [Function],
            "setContext": [Function],
          },
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "runSaga": Object {
              "@@redux-saga/TASK": true,
              "cancel": [Function],
              "cont": undefined,
              "done": Promise {},
              "error": [Function],
              "id": 1,
              "isAborted": [Function],
              "isCancelled": [Function],
              "isRunning": [Function],
              "joiners": Array [],
              "name": "root",
              "result": [Function],
              "setContext": [Function],
            },
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
